public/achievementInfo.php

```diff
@@ -85,13 +91,71 @@
 
 <body>
 <?php RenderHeader($userDetails); ?>
-<?php if ($permissions >= Permissions::Developer): ?>
+<?php if (($permissions >= Permissions::Developer || ($isSoleAuthor && $permissions >= Permissions::JuniorDeveloper && $achFlags === AchievementType::Unofficial))): ?>
     <script>
+        function updateAchievementDetails() {
+        var title = $('#titleinput').val();
+        var description = $('#descriptioninput').val();
+        var points = $('#pointsinput').val();
+        var validPoints = <?php echo json_encode(VALID_POINTS); ?>;
+        var titleLength = new Blob([title]).size;
+        var descLength = new Blob([descLength]).size;
+        var maxTitle = <?php echo MAX_TITLE; ?>;
+        var maxDescription = <?php echo MAX_DESCRIPTION; ?>;
+        showStatusMessage('Updating...');
+        if ( titleLength > maxTitle) {
+                showStatusFailure('Error: Title too long');
+                return;
+        } else if ( descLength > maxDescription) {
+                showStatusFailure('Error: Description too long');
+                return;
+        } else if ( points == '' || !validPoints.includes(Number(points))) {
+                showStatusFailure('Error: Invalid Points');
+                return;
+        }
```

Comment:
maybe being picky, but trying to improve readability/clarity...

those `else`s are not necessary when the previous `if`s are `return`ing.

Doing like in the example would have the same result and a reduced cognitive load (fancy words for "less code to read and understand"):
```php
if(condition1) {
  doSomething1();
  return;
}
if(condition2) {
  doSomething2();
  return;
} 
if(condition3) {
  doSomething3();
  return;
}
```